### DISCUSSION TITLE
Updating the projection and view matrix for webXR

### DISCUSSION THREAD

**Discussion by positlabs** [cables-gl/cables_docs#165] [Answers & Questions]
I'm trying to get a webxr session working in cables, but I'm having trouble getting the view/projection matrices to update correctly. The cube remains in the center of the screen, not tracked to the world. Any ideas what I'm doing wrong here?

https://cables.gl/edit/Hs_uFX

```
const onXRFrame = (time, frame) => {
    // Queue up the next draw request.
    session.requestAnimationFrame(onXRFrame)

    // Bind the graphics framebuffer to the baseLayer's framebuffer
    gl.bindFramebuffer(gl.FRAMEBUFFER, session.renderState.baseLayer.framebuffer)
    
    // Retrieve the pose of the device.
    const pose = frame.getViewerPose(referenceSpace)
    if (pose) {
        const view = pose.views[0]
    
        const viewport = session.renderState.baseLayer.getViewport(view)
        op.patch.cgl.setSize(viewport.width, viewport.height)
    
        // Use the view's transform matrix and projection matrix to configure the camera
        
        // set projection matrix
        cgl.pushPMatrix()
        mat4.copy(cgl.pMatrix, view.projectionMatrix)
        cgl.popPMatrix()
        
        // view transform
        cgl.pushViewMatrix()
        mat4.multiply(cgl.vMatrix, cgl.vMatrix, view.transform.matrix)
        cgl.popViewMatrix()
        
        outTrigger.trigger()
    }
}
```

**Comment by finofinissimo** [comment]
Instead of the ViewProjection, shouldnt you be multiply by ModelViewProjection?

**Reply by positlabs** [reply]
I pulled this from the threejs example and they are only setting those values on the camera. https://developers.google.com/ar/develop/webxr/hello-webxr

I'm just not sure how to do something like this in cables. Right now it seems like the matrices are just flipping the sides on the material or something. It's not positioned in space at all. I have a live demo up here: https://positlabs.github.io/cables-webxr/

It seems like the projection matrix is just used for VR to differentiate between each eye, so maybe I can skip that step here. 

"The projection matrix that will transform the scene to appear correctly given the point-of-view indicated by eye." https://developer.mozilla.org/en-US/docs/Web/API/XRView
